lib/rubygems/specification_policy.rb

```diff
@@ -173,13 +173,18 @@ def validate_duplicate_dependencies # :nodoc:
   end
 
   ##
+  # Checks that dependencies are not self referencing itself.
```

Comment:
This sentence feels weird to me, "self" and "itself" seem to add some redundancy. But I don't have a great alternative, maybe

```suggestion
  # Checks that no dependencies reference the gem itself.
```

but doesn't feel great it. A native speaker should word this better.

Reply:
Is the meaning here that the gem shouldn't refer to itself?

Reply:
> Or even "strongly discouraged" instead of "not recommended".

Yes, like this one https://rubygems.org/gems/standalone_migrations/versions/2.0.4.

Reply:
I like this

```suggestion
  # Checks that the gem does not depend on itself.
```

Reply:
Yep, I like that also. 👍🏻